File: AngularBoot/src/app/pages/_models/auth/registerRequest.ts
/**
 * POJO to creaet Insert new user into LDAP server
 */
export class RegisterRequest {
    username: string;
    password: string;
    role: string;

    public constructor() { }

    setUsername(username: string) {
        this.username = username;
    }

    getUserName(): string {
        return this.username;
    }

    setPassword(username: string) {
        this.password = username;
    }

    getPassword(): string {
        return this.password;
    }

    setUserRole(role: string) {
        this.role = role;
    }

    getUserRole(): string {
        return this.role;
    }
}

File: SpringBoot/shop-founders-modules/shop-founders-core/src/main/java/hf/shopfounders/model/RegisterRequest.java
package hf.shopfounders.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import hf.shopfounders.exception.BaseErrorCode;
import hf.shopfounders.validation.ParamNotNull;

public class RegisterRequest {

    @ParamNotNull
    @JsonProperty(value = "username")
    private String username;
    @ParamNotNull
    @JsonProperty(value = "password")
    private String password;
    @ParamNotNull
    @JsonProperty(value = "role")
    private String role;

    public RegisterRequest() {
    }

    public RegisterRequest(String username, String password, String role) {
        this.username = username;
        this.password = password;
        this.role = role;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @Override
    public String toString() {
        return String.format(
                "RegisterRequest[username='%s', password='%s',role='%s']",
                username, password, role);
    }
}
